platforms/gke/base/use-cases/federated-learning/examples/cross-device/setup-environment.sh

```diff
@@ -0,0 +1,142 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Don't set errexit because we might source this script from an interactive
+# shell, and we don't want to exit the shell on errors
+# set -o errexit
+set -o nounset
+set -o pipefail
+
+if [[ ! -v ACP_REPO_DIR ]]; then
+  SCRIPT_DIRECTORY_PATH="$(cd -P "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
+
+  echo "This script directory path is: ${SCRIPT_DIRECTORY_PATH}"
+
+  ACP_REPO_DIR="$(readlink -f "${SCRIPT_DIRECTORY_PATH}/../../../../../../../")"
+  export ACP_REPO_DIR
+  export ACP_PLATFORM_BASE_DIR="${ACP_REPO_DIR}/platforms/gke/base"
+
+  echo "ACP_REPO_DIR: ${ACP_REPO_DIR}"
+  echo "ACP_PLATFORM_BASE_DIR: ${ACP_PLATFORM_BASE_DIR}"
+
+  export ACP_PLATFORM_CORE_DIR="${ACP_PLATFORM_BASE_DIR}/core"
+  echo "ACP_PLATFORM_CORE_DIR: ${ACP_PLATFORM_CORE_DIR}"
+fi
+
+# shellcheck disable=SC1091
+source "${ACP_PLATFORM_BASE_DIR}/use-cases/federated-learning/common.sh"
+
+echo "Setting up the environment for the cross-device federated learning example"
+CROSS_DEVICE_EXAMPLE_TENANT_NAME="fl-1"
+CROSS_DEVICE_MODEL_BUCKET="model-0"
+CROSS_DEVICE_AGGREGATED_GRADIENT_BUCKET="aggregated-gradient-0"
+CROSS_DEVICE_CLIENT_GRADIENT_BUCKET="client-gradient-0"
+CROSS_DEVICE_AGGREGATOR_SERVICE_ACCOUNT="aggregator"
+CROSS_DEVICE_MODELUPDATER_SERVICE_ACCOUNT="modelupdater"
+CROSS_DEVICE_COLLECTOR_SERVICE_ACCOUNT="collector"
+CROSS_DEVICE_TASK_ASSIGNMENT_SERVICE_ACCOUNT="task-assignment"
+CROSS_DEVICE_TASK_MANAGEMENT_SERVICE_ACCOUNT="task-management"
+CROSS_DEVICE_TASK_SCHEDULER_SERVICE_ACCOUNT="task-scheduler"
+CROSS_DEVICE_TASK_BUILDER_SERVICE_ACCOUNT="task-builder"
+CROSS_DEVICE_AGGREGATOR_IMAGE="europe-docker.pkg.dev/driven-density-457716-m7/container-image-repository/aggregator_image@sha256:1043eb980b618e325d5b490c879fc72f211a70104d7f645b1bbb70996a42de2c"
+CROSS_DEVICE_MODELUPDATER_IMAGE="europe-docker.pkg.dev/driven-density-457716-m7/container-image-repository/model_updater_image@sha256:b0bc213e4cb34c99525345b1d544371ce5c9d647ec08405a9ca96d61e0b272fa"
+CROSS_DEVICE_COLLECTOR_IMAGE="europe-docker.pkg.dev/driven-density-457716-m7/container-image-repository/collector_image"
+CROSS_DEVICE_TASK_ASSIGNMENT_IMAGE="europe-docker.pkg.dev/driven-density-457716-m7/container-image-repository/task_assignment_image"
+CROSS_DEVICE_TASK_MANAGEMENT_IMAGE="europe-docker.pkg.dev/driven-density-457716-m7/container-image-repository/task_management_image"
+CROSS_DEVICE_TASK_SCHEDULER_IMAGE="europe-docker.pkg.dev/driven-density-457716-m7/container-image-repository/task_scheduler_image"
+CROSS_DEVICE_TASK_BUILDER_IMAGE="europe-docker.pkg.dev/driven-density-457716-m7/container-image-repository/task_builder_image"
```

Comment:
We should not hardcode these URLs because they introduce a dependency. I suppose you need to build those images using Cloud Build, maybe?